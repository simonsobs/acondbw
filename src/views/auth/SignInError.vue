<template>
  <v-container fill-height fluid>
    <v-row align="center" justify="center">
      <v-card flat min-width="250px">
        <v-card-title>Sign In Unsuccessful</v-card-title>
        <v-card-text v-if="error">
          <v-alert type="error" outlined>
            <strong v-if="error.error">{{ error.error }}:</strong>
            <span v-if="error.error_description">
              {{ error.error_description }}</span
            ><v-btn
              v-if="error.error_uri"
              icon
              small
              color="error"
              :href="error.error_uri"
              target="_blank"
              ><v-icon small>mdi-open-in-new</v-icon></v-btn
            >
          </v-alert>
        </v-card-text>
        <v-card-actions>
          <v-btn outlined to="/">Cancel</v-btn>
          <v-spacer></v-spacer>
          <v-btn outlined @click="tryAgain">Try again</v-btn>
        </v-card-actions>
      </v-card>
    </v-row>
  </v-container>
</template>

<script>
export default {
  name: "SignInError",
  computed: {
    error() {
      const auth = this.$store.state.auth;
      if (auth) {
        return auth.lastError;
      } else {
        return null;
      }
    },
  },
  methods: {
    tryAgain() {
      this.$router.push({ name: "SignIn" });
    },
  },
};
</script>
