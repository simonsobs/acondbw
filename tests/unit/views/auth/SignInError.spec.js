import Vue from "vue";
import VueRouter from "vue-router";
import Vuex from "vuex";
import Vuetify from "vuetify";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";

import SignInError from "@/views/auth/SignInError.vue";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("SignInError.vue", () => {
  let localVue;
  let vuetify;

  function createWrapper(store) {
    return shallowMount(SignInError, {
      localVue,
      vuetify,
      store,
    });
  }

  beforeEach(function () {
    localVue = createLocalVue();
    localVue.use(Vuex);
    vuetify = new Vuetify();
  });

  it("match snapshot without error", async () => {
    const store = new Vuex.Store({});
    const wrapper = createWrapper(store);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("match snapshot with error", async () => {
    const store = new Vuex.Store({
      state: {
        auth: {
          lastError: {
            error: "access denied",
            error_description: "The user has denied your application access.",
            error_uri:
              "https://docs.github.com/apps/managing-oauth-apps/troubleshooting-authorization-request-errors/#access-denied",
          },
        },
      },
    });
    const wrapper = createWrapper(store);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
