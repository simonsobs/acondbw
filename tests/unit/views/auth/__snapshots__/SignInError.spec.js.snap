// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInError.vue match snapshot with error 1`] = `
<v-container-stub tag="div" fluid="true" fill-height="">
  <v-row-stub tag="div" align="center" justify="center">
    <v-card-stub loaderheight="4" tag="div" minwidth="250px" flat="true">
      <v-card-title-stub>Sign In Unsuccessful</v-card-title-stub>
      <v-card-text-stub>
        <v-alert-stub outlined="true" tag="div" value="true" closelabel="$vuetify.close" closeicon="$cancel" icon="" type="error"><strong>access denied:</strong> <span>
            The user has denied your application access.</span>
          <v-btn-stub color="error" tag="button" activeclass="" href="https://docs.github.com/apps/managing-oauth-apps/troubleshooting-authorization-request-errors/#access-denied" target="_blank" small="true" icon="true" type="button">
            <v-icon-stub small="">mdi-open-in-new</v-icon-stub>
          </v-btn-stub>
        </v-alert-stub>
      </v-card-text-stub>
      <v-card-actions-stub>
        <v-btn-stub outlined="true" tag="button" activeclass="" to="/" type="button">Cancel</v-btn-stub>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub outlined="true" tag="button" activeclass="" type="button">Try again</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-row-stub>
</v-container-stub>
`;

exports[`SignInError.vue match snapshot without error 1`] = `
<v-container-stub tag="div" fluid="true" fill-height="">
  <v-row-stub tag="div" align="center" justify="center">
    <v-card-stub loaderheight="4" tag="div" minwidth="250px" flat="true">
      <v-card-title-stub>Sign In Unsuccessful</v-card-title-stub>
      <!---->
      <v-card-actions-stub>
        <v-btn-stub outlined="true" tag="button" activeclass="" to="/" type="button">Cancel</v-btn-stub>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub outlined="true" tag="button" activeclass="" type="button">Try again</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-row-stub>
</v-container-stub>
`;
